.github/workflows/test-backend.yml

```diff
@@ -0,0 +1,47 @@
+name: Test Backend
+on: 
+  pull_request:
+
+jobs:
+  testing:
+    name: Testing Backend
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./backend
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install pytest-md pytest-emoji
```

Comment:
should these two be in our requirements file, or are they github action only requirements?

Reply:
these are only needed by the `pytest-action@v2` GitHub action to create a report in markdown and add emojis to the report.